Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm

```diff
@@ -390,11 +391,49 @@ - (CFURLRef)_cfurl;
     return { String(), false };
 }
 
+Vector<Cookie> NetworkStorageSession::vectorOfCookiesForSession(const URL& firstParty, const SameSiteInfo& sameSiteInfo, const URL& url, std::optional<FrameIdentifier> frameID, std::optional<PageIdentifier> pageID, IncludeHTTPOnlyOrNot includeHTTPOnly, IncludeSecureCookies includeSecureCookies, ApplyTrackingPrevention applyTrackingPrevention, ShouldRelaxThirdPartyCookieBlocking shouldRelaxThirdPartyCookieBlocking, const CookieStoreGetOptions& cookieStoreGetOptions) const
+{
+    ASSERT(hasProcessPrivilege(ProcessPrivilege::CanAccessRawCookies) || m_isInMemoryCookieStore);
+
+    BEGIN_BLOCK_OBJC_EXCEPTIONS
+
+    auto cookies = cookiesForURL(firstParty, sameSiteInfo, url, frameID, pageID, applyTrackingPrevention, shouldRelaxThirdPartyCookieBlocking);
+    if (![cookies count])
+        return { };
+
+    Vector<Cookie> cookiesVector;
+    for (NSHTTPCookie *cookie in cookies.get()) {
+        if (![[cookie name] length])
+            continue;
+        if (!includeHTTPOnly && [cookie isHTTPOnly])
+            continue;
+        if ([cookie isSecure]) {
+            if (includeSecureCookies == IncludeSecureCookies::No)
+                continue;
+        }
+
+        auto convertedCookie = Cookie(cookie);
+        if (convertedCookie.name != cookieStoreGetOptions.name)
```

Comment:
It is too wasteful to be doing this check *after* constructing the WebCore::Cookie object. You should do the check on the platform cookie type instead.

Reply:
Also, cookieStoreGetOptions.name can be null, in which case we want to return ALL cookies. The current check is wrong.